src/Radicle/Internal/Doc.hs

```diff
@@ -0,0 +1,191 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE TemplateHaskell       #-}
+
+module Radicle.Internal.Doc where
+
+import           Protolude hiding (Any)
+
+import           Codec.Serialise (Serialise(..))
+import           Data.Copointed (Copointed(..))
+import qualified Data.Default as Default
+import           Data.Pointed (Pointed(..))
+import           Language.Haskell.TH.Quote
+import           Text.Pandoc
+
+type Blocks = [Block]
+type Inlines = [Inline]
+
+data Pan = Blocks Blocks | Inlines Inlines
+  deriving (Eq, Ord, Show, Read)
+
+instance Semigroup Pan where
+  Blocks x <> Blocks y = Blocks (x <> y)
+  Inlines x <> Inlines y = Inlines (x <> [Space] <> y)
+  Blocks x <> Inlines y = Blocks (x <> [Para y])
+  Inlines x <> Blocks y = Blocks (Para x : y)
+
+class ToPan a where
+  pan :: a -> Pan
+
+instance ToPan Pan where
+  pan = identity
+
+blocks :: ToPan a => a -> Blocks
+blocks x = case pan x of
+  Blocks bs  -> bs
+  Inlines is -> [Para is]
+
+instance ToPan Blocks where
+  pan = Blocks
+
+instance ToPan a => ToPan (Maybe a) where
+  pan Nothing  = Inlines []
```

Comment:
This seems a little arbitrary. Why not `Blocks []` for instance?

(Is this class all that useful/sensible, rather than having multiple functions?)

Reply:
`Inlines []` makes it so that when combined with a piece of inline documentation, it doesn't promote to a block.